### DISCUSSION TITLE
Add `quarto-prerelease` to `homebrew/cask-versions`

### DISCUSSION THREAD

**Discussion by gvelasq** [quarto-dev/quarto-cli#2955] [Q&A]
Hello, I use Homebrew to install and upgrade Quarto.

Would it be possible to add a repository for `quarto-prerelease` in `homebrew/cask-versions`? This recently came up in https://github.com/quarto-ext/lightbox/issues/14 where having the prerelease version 1.2 was the workaround to correctly render `example.qmd` in the `quarto-ext/lightbox` repository. The proposed workflow to install a prerelease version would be:

```bash
brew tap homebrew/cask-versions
brew install --cask quarto-prerelease
```

**Comment by cscheid** [comment]
We're not the maintainers of the `homebrew` recipes for quarto, so you'd have to reach out to them instead.

**Comment by mcanouil** [comment]
If it helps, you can write your own local recipe.
You simply need to update the version to whatever version you want and update the sha256 using the macos.pkg sha listed in the `quarto-1.2.231-checksums.txt` file of every release (<https://github.com/quarto-dev/quarto-cli/releases>).

- `quarto.rb`
  ```rb
  cask "quarto" do
    version "1.2.231"
    sha256 "b5d73eb31693cb0ac4d9ae68be06b2b0d984b838eaa2102621ec365da2f926c5"

    url "https://github.com/quarto-dev/quarto-cli/releases/download/v#{version}/quarto-#{version}-macos.pkg",
        verified: "github.com/quarto-dev/quarto-cli/"
    name "quarto"
    desc "Scientific and technical publishing system built on Pandoc"
    homepage "https://www.quarto.org/"

    livecheck do
      url :url
      strategy :github_latest
    end

    depends_on macos: ">= :el_capitan"

    pkg "quarto-#{version}-macos.pkg"

    uninstall pkgutil: "org.rstudio.quarto"

    zap trash: "~/Library/Caches/quarto"
  end
  ```
- (re)Install Cask
  ```
  brew reinstall --cask quarto.rb
  ```

To note, `Quarto` will appear as outdated in homebrew, but you'll get the pre-release.
You can revert to the latest stable version with `brew upgrade quarto` and/or `brew reinstall --cask quarto`.

Note: here I overwrite the default `quarto` recipe to avoid conflicts or having two version of Quarto.

**Comment by gvelasq** [comment]
Thank you, both.

It looks like @rundel submitted the first commit for `quarto` in `homebrew/cask` at https://github.com/Homebrew/homebrew-cask/pull/118002, I will investigate how to submit `quarto-prerelease` formally to `homebrew/cask-versions`, but would appreciate any help along the way from @rundel or others, time and interest permitting!

**Reply by mcanouil** [reply]
For this cask `quarto-prerelease` (most likely a copy of the original cask), the proper location should be <https://github.com/Homebrew/homebrew-cask-versions>.
The question is who's going to maintain the two 😅

**Comment by rundel** [comment]
I don't have any specific advice - the formula should be nearly identical to the existing cask, just point it at the prerelease version and I think the livecheck block can be removed so it uses the latest release (see https://github.com/rundel/homebrew-quarto-cli for a formula that uses the prerelease).
